### PR TITLE
ruby2.5: Update version from 2.5.5-3+deb10u7 to 2.5.5-3+deb10u9

### PR DESCRIPTION
**Pull request by mozoko**
# Purpose of pull request

## 2.5.5-3+deb10u8

This version fixes the following CVEs.

- CVE-2024-35176
- CVE-2024-39908
- CVE-2024-41123
- CVE-2024-41946
- CVE-2024-43398
- CVE-2024-49761

ELA: ELA-1305-1

## 2.5.5-3+deb10u9
This version fixes a regression in the REXML gem. Some valid XML file
were wrongly considered invalid for some namespace corner case.

ELA: ELA-1305-2

# Test

1. Build package
2. Run image and ruby ptest on the qemuarm64 environment

## How to test the package
1. Build qemuarm64 image
2. Run qemu image
3. Prepare ruby ptest
4. Run ruby ptest

### Build qemuarm64 image
Add the following to local.conf.

```
MACHINE = "qemuarm64"
IMAGE_INSTALL_append = " ruby ruby-ptest"
CORE_IMAGE_EXTRA_INSTALL += " kernel-modules"
IMAGE_INSTALL_append = " kmod"
```

And build the qemuarm64 image.

```
$ bitbake core-image-minimal
```

### Run qemu image
```
runqemu qemuarm64 nographic qemuparams="-smp 8 -m 8192"
```

### Prepare ruby ptest

"test_deadlock_by_signal_at_forking" test may fail because of the computer resources lack on which it runs.
It can be avoid by increasing the timeout duration.
The difference of increasing timeouts is as following.

```diff
--- /usr/lib/ruby/ptest/test/ruby/test_process.rb.original
+++ /usr/lib/ruby/ptest/test/ruby/test_process.rb
@@ -2167,7 +2167,7 @@
   end

   def test_deadlock_by_signal_at_forking
-    assert_separately(["-", RUBY], <<-INPUT, timeout: 80)
+    assert_separately(["-", RUBY], <<-INPUT, timeout: 100)
       ruby = ARGV.shift
       GC.start # reduce garbage
       GC.disable # avoid triggering CoW after forks
```

### Run ruby ptest

```
# ptest-runner -t 7200 ruby | tee "ruby-test.log"
```

# Test result
## Build package
Build succeeded.

```
$ bitbake ruby
Loading cache: 100% |########################################################################################################################| Time: 0:00:00Loaded 2382 entries from dependency cache.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "1.42.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "universal"
TARGET_SYS           = "aarch64-emlinux-linux"
MACHINE              = "qemuarm64"
DISTRO               = "emlinux"
DISTRO_VERSION       = "2.10"
TUNE_FEATURES        = "aarch64 armv8a crc"
TARGET_FPU           = ""
meta
meta-yocto-bsp       = "HEAD:d4b57c68b22027c2bedff335dee06af963e4f8a8"
meta-debian          = "HEAD:05fd40d23535cb8b9ccc8dca40950900edd43933"
meta-debian-extended = "ruby-update:761af7501ce4c9a9c0d5c452ef0dcfbae3c91dc5"
meta-emlinux         = "HEAD:4e028ff087130d048397fd4d060016a6acf47833"
meta-emlinux-private = "HEAD:0240322695096c9406f4cd9cd3754f311489bab5"

Initialising tasks: 100% |###################################################################################################################| Time: 0:00:00Sstate summary: Wanted 6 Found 0 Missed 6 Current 467 (0% match, 98% complete)
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks
NOTE: Tasks Summary: Attempted 2178 tasks of which 2159 didn't need to be rerun and all succeeded.
```
## Package test
Only the same tests as the previous version failed. No regression. (see #497 )

```
$ tail -n 12 new-ruby-test.log

Finished
--------
   Tests executed: 762
             PASS: 748
             FAIL: 0
EXPECTED FAILURES: 14

DURATION: 4363
END: /usr/lib/ruby/ptest
2025-02-28T08:27
STOP: ptest-runner
```

```
$ grep '^FAIL' new-ruby-test.log
FAIL: (EXPECTED) test/mkmf/test_constant.rb
FAIL: (EXPECTED) test/mkmf/test_convertible.rb
FAIL: (EXPECTED) test/mkmf/test_flags.rb
FAIL: (EXPECTED) test/mkmf/test_have_func.rb
FAIL: (EXPECTED) test/mkmf/test_have_library.rb
FAIL: (EXPECTED) test/mkmf/test_have_macro.rb
FAIL: (EXPECTED) test/mkmf/test_signedness.rb
FAIL: (EXPECTED) test/mkmf/test_sizeof.rb
FAIL: (EXPECTED) test/ripper/test_files.rb
FAIL: (EXPECTED) test/ruby/test_fiber.rb
FAIL: (EXPECTED) test/ruby/test_io.rb
FAIL: (EXPECTED) test/rubygems/test_gem.rb
FAIL: (EXPECTED) test/rubygems/test_gem_commands_environment_command.rb
FAIL: (EXPECTED) test/rubygems/test_gem_commands_update_command.rb
```

### CODE REVIEW

**Comment by hiraku-wfs** [comment]
Thanks!
